### PR TITLE
Added qb-target and polyzone and removed Drawtext3D (IDK How it was reverted)

### PR DESCRIPTION
**Pull request by amantu-qbit**

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -199,40 +246,79 @@ CreateThread(function()
...
-				elseif #(pos - vector3(Config.Locations["freedom"].coords.x, Config.Locations["freedom"].coords.y, Config.Locations["freedom"].coords.z)) < 2.5 then
-					DrawText3D(Config.Locations["freedom"].coords.x, Config.Locations["freedom"].coords.y, Config.Locations["freedom"].coords.z, "Check time")
-				end
+freedom = BoxZone:Create(vector3(Config.Locations["freedom"].coords.x, Config.Locations["freedom"].coords.y, Config.Locations["freedom"].coords.z), 2.75, 2.75, {
+	name="freedom",
+	debugPoly = false,
+})
+freedom:onPlayerInOut(function(isPointInside)
+	if isPointInside then
+		exports['qb-drawtext']:DrawText('[E] Check Time', 'left')
```
Comment (GhzGarage) [inline client/main.lua:255]: Wrong export name

File: client/main.lua [hunk 2]
```diff
@@ -199,40 +246,79 @@ CreateThread(function()
...
-				elseif #(pos - vector3(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z)) < 10 then
-					DrawMarker(2, Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.3, 0.2, 255, 55, 22, 222, false, false, false, 1, false, false, false)
-				end
+canteen = BoxZone:Create(vector3(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z), 2.75, 7.75, {
+name="canteen",
+debugPoly = false,
+})
+canteen:onPlayerInOut(function(isPointInside)
+	if isPointInside then
+		exports['qb-drawtext']:DrawText('[E] Open Canteen', 'left')
```
Comment (GhzGarage) [inline client/main.lua:269]: Wrong export name

File: client/main.lua [hunk 3]
```diff
@@ -199,40 +246,79 @@ CreateThread(function()
...
+		insidecanteen = true
+	else
+		insidecanteen = false
+		exports['qb-drawtext']:HideText()
+	end
+end)
+
+local function interaction()
+	if Config.UseTarget then
+			print(freedom_ped, canteen_ped)
```
Comment (GhzGarage) [inline client/main.lua:279]: Leftover print

Comment (amantu-qbit) [reply client/main.lua:279]: Resolving. Added this export before drawtext was merged with core

File: client/jobs.lua [hunk 4]
```diff
@@ -45,41 +45,80 @@ local function JobDone()
...
+        })
+    else
+        CreateThread(function()
+            local electricityzone = BoxZone:Create(vector3(Config.Locations.jobs[currentJob][currentLocation].coords.x, Config.Locations.jobs[currentJob][currentLocation].coords.y, Config.Locations.jobs[currentJob][currentLocation].coords.z), 3.0, 5.0, {
+                name="electricity",
+                debugPoly=false,
+            })
+            electricityzone:onPlayerInOut(function(isPointInside)
+                if isPointInside then
+                    exports['qb-drawtext']:DrawText('[E] Electricity Work', 'left')
```
Comment (GhzGarage) [inline client/jobs.lua:75]: Wrong export name

File: client/main.lua [hunk 5]
```diff
@@ -84,8 +72,59 @@ RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
 	end)
 
 	PlayerJob = QBCore.Functions.GetPlayerData().job
+	RequestModel("s_m_m_armoured_01")
```
Comment (GhzGarage) [inline client/main.lua:75]: Why are you spawning the peds on player login and resource restart even if user is not using target? There is an export that you can use for this

Comment (amantu-qbit) [reply client/main.lua:75]: Using ped as indication instead of DrawMarker when Target is false. And using ped as target entity when Target is true

Comment (GhzGarage) [reply client/main.lua:75]: Ok

**Comment by Andyauk** [comment]
hopefully the updates posted will fix the F8 spam, too.

<a href="https://imgbb.com/">Untitled</a>

**Pull request merged**